sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt

```diff
@@ -182,27 +178,24 @@ internal class ViewAnnotationManagerImpl(
   }
 
   private fun drawAnnotationViews(
-    positionDescriptorList: List<ViewAnnotationPositionDescriptor>
+    positionDescriptorCoreList: List<ViewAnnotationPositionDescriptor>
   ) {
-    positionDescriptorMap.clear()
-    idsToRepositionSet.clear()
-    idsToAddSet.clear()
-    idsToDeleteSet.clear()
-
-    positionDescriptorList.forEachIndexed { i, descriptor ->
-      positionDescriptorMap[descriptor.identifier] =
+    this.positionDescriptorList.clear()
+    // TODO rewrite to avoid spawning collections
+    val sortedKeyList = positionDescriptorCoreList.map { it.identifier }
+    val idsToAddSet = sortedKeyList.minus(currentViewsDrawnMap.keys)
+    val idsToDeleteSet = currentViewsDrawnMap.keys.minus(sortedKeyList)
+    positionDescriptorCoreList.forEach { descriptor ->
+      positionDescriptorList.add(
```

Comment:
Could we just call the function that adds view and brings it to front from https://github.com/mapbox/mapbox-maps-android/pull/835/files#diff-734ba43e94cb758b50aaa216439b68bffe47a7568f40aebfe3c27dcc4d3aeccbR210 instead of second iteration over `positionDescriptorList` below?

Reply:
nice, got your idea 👍